Mathlib/RingTheory/KrullDimension/Basic.lean

```diff
@@ -80,14 +82,30 @@ lemma Ring.krullDimLE_zero_iff : Ring.KrullDimLE 0 R ↔ ∀ I : Ideal R, I.IsPr
 lemma Ring.KrullDimLE.mk₀ (H : ∀ I : Ideal R, I.IsPrime → I.IsMaximal) : Ring.KrullDimLE 0 R := by
   rwa [Ring.krullDimLE_zero_iff]
 
-instance (priority := 100) Ideal.isMaximal_of_isPrime [Ring.KrullDimLE 0 R]
-    (I : Ideal R) [I.IsPrime] : I.IsMaximal :=
+lemma Ideal.isMaximal_of_isPrime [Ring.KrullDimLE 0 R] (I : Ideal R) [I.IsPrime] : I.IsMaximal :=
   Ring.krullDimLE_zero_iff.mp ‹_› I ‹_›
 
+instance (priority := 100) (I : Ideal R) [I.IsPrime] [Ring.KrullDimLE 0 R] : I.IsMaximal :=
+  I.isMaximal_of_isPrime
```

Comment:
@erdOne I changed the argument order of this instance and kept the original order as a lemma, because this is better order for typeclass search according to [#21449](https://github.com/leanprover-community/mathlib4/pull/21449/files).